dp-hello-world-event/service/service.go

```diff
@@ -0,0 +1,165 @@
+package service
+
+import (
+	"context"
+	"github.com/ONSdigital/dp-hello-world-event/config"
+	"github.com/ONSdigital/dp-hello-world-event/event"
+	"github.com/ONSdigital/log.go/log"
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+	"time"
+)
+
+// Service contains all the configs, server and clients to run the event handler service
+type Service struct {
+	server          HTTPServer
+	router          *mux.Router
+	serviceList     *ExternalServiceList
+	healthCheck     HealthChecker
+	consumer        KafkaConsumer
+	eventConsumer   EventConsumer
+	shutdownTimeout time.Duration
+}
+
+// Run the service
+func Run(ctx context.Context, serviceList *ExternalServiceList, buildTime, gitCommit, version string, svcErrors chan error) (*Service, error) {
+	log.Event(ctx, "running service", log.INFO)
+
+	// Read config
+	cfg, err := config.Get()
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to retrieve service configuration")
+	}
+	log.Event(ctx, "got service configuration", log.Data{"config": cfg}, log.INFO)
+
+	// Get HTTP Server with collectionID checkHeader middleware
+	r := mux.NewRouter()
+	s := serviceList.GetHTTPServer(cfg.BindAddr, r)
+
+	// Get Kafka consumer
+	consumer, err := serviceList.GetKafkaConsumer(ctx, cfg)
+	if err != nil {
+		log.Event(ctx, "failed to initialise kafka consumer", log.FATAL, log.Error(err))
+		return nil, err
+	}
+
+	// Event Handler for Kafka Consumer with the created S3 Clients and Vault
+	eventConsumer := event.NewConsumer()
+	eventConsumer.Consume(ctx, consumer, &event.HelloCalledHandler{})
+
+	// Get HealthCheck
+	hc, err := serviceList.GetHealthCheck(cfg, buildTime, gitCommit, version)
+	if err != nil {
+		log.Event(ctx, "could not instantiate healthcheck", log.FATAL, log.Error(err))
+		return nil, err
+	}
+
+	if err := registerCheckers(ctx, hc, consumer); err != nil {
+		return nil, errors.Wrap(err, "unable to register checkers")
+	}
+
```

Comment:
I would suggest adding the Kafka logger go-routine here, after having initialised all dependencies (example https://github.com/ONSdigital/dp-kafka/blob/main/examples/consumer-concurrent/main.go#L92):
```
consumer.Channels().LogErrors(ctx, "Consumer error")
```